fiftyone/core/utils.py

```diff
@@ -3143,3 +3143,45 @@ def validate_hex_color(value):
 
 
 fos = lazy_import("fiftyone.core.storage")
+
+
+@contextmanager
+def async_executor(
+    *, max_workers, skip_failures=False, warning="Async failure"
+):
+    """
+    Context manager that provides a function for submitting tasks to a thread
+    pool executor. All tasks are joined when the context is exited.
+
+    Example:
```

Comment:
@exupero FYI in order for your example code to be rendered as a code block in the method's docstring at docs.voxel51.com, you need to use a double colon `Example::` (**unlike** `Args:` and `Returns:`, which are special Sphinx keywords).

For example: 
- source: https://github.com/voxel51/fiftyone/blame/2853b72cb393c1b08d1f14bcfd06da5ee7d3ae50/fiftyone/core/utils.py#L736-L744
- rendered: https://docs.voxel51.com/api/fiftyone.core.utils.html#fiftyone.core.utils.lazy_import